src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -230,19 +223,34 @@ protected override bool Receive(object message)
         {
             switch (message)
             {
-                case KafkaConsumerActorMetadata.Internal.Commit commit when _rebalanceInProgress:
-                    _rebalanceCommitStash = _rebalanceCommitStash.Union(commit.Offsets);
-                    _rebalanceCommitSenders = _rebalanceCommitSenders.Add(Sender);
-                    return true;
-
                 case KafkaConsumerActorMetadata.Internal.Commit commit:
-                    _commitRefreshing.Add(commit.Offsets);
-                    var replyTo = Sender;
-                    Commit(commit.Offsets, msg => replyTo.Tell(msg));
+                    // prepending as later received offsets are most likely higher
+                    _commitMaps = ImmutableList<TopicPartitionOffset>.Empty
+                        .Add(commit.TopicPartitionOffset)
+                        .AddRange(_commitMaps);
+                    _commitSenders = _commitSenders.Add(Sender);
                     return true;
 
-                case Internal.Poll<K, V> poll:
-                    ReceivePoll(poll);
+                case KafkaConsumerActorMetadata.Internal.CommitWithoutReply commitWithoutReply:
```

Comment:
`CommitWithoutReply` is fire-and-forget committing - usually happens during an emergency commit when things are shutting down.